httpserver-rs/bin/main.rs

```diff
@@ -42,36 +48,76 @@ impl ProviderHandler for HttpServerProvider {
         let settings =
             load_settings(&ld.values).map_err(|e| RpcError::ProviderInit(e.to_string()))?;
 
-        let http_server = HttpServerCore::new(settings.clone(), get_host_bridge());
-        http_server.start(ld.clone()).await.map_err(|e| {
+        let http_server = HttpServerCore::new(
+            settings.clone(),
+            get_host_bridge().lattice_prefix().to_string(),
+            call_actor,
+        );
+
+        http_server.start(ld).await.map_err(|e| {
             RpcError::ProviderInit(format!(
                 "starting httpserver for {} {:?}: {}",
                 &ld.actor_id, &settings.address, e
             ))
         })?;
 
-        let mut update_map = self.actors.write().await;
-        update_map.insert(ld.actor_id.to_string(), http_server);
+        self.actors.insert(ld.actor_id.to_string(), http_server);
 
         Ok(true)
     }
 
     /// Handle notification that a link is dropped - stop the http listener
     async fn delete_link(&self, actor_id: &str) {
-        let mut aw = self.actors.write().await;
-        if let Some(server) = aw.remove(actor_id) {
+        if let Some(entry) = self.actors.remove(actor_id) {
             tracing::info!(%actor_id, "httpserver stopping listener for actor");
-            server.begin_shutdown().await;
+            entry.1.begin_shutdown();
         }
     }
 
     /// Handle shutdown request by shutting down all the http server threads
     async fn shutdown(&self) -> Result<(), Infallible> {
-        let mut aw = self.actors.write().await;
         // empty the actor link data and stop all servers
-        for (_, server) in aw.drain() {
-            server.begin_shutdown().await;
-        }
+        self.actors.clear();
         Ok(())
     }
 }
+
+/// forward HttpRequest to actor.
+//#[instrument(level = "debug", skip(_lattice_id, ld, req, timeout), fields(actor_id = %ld.actor_id))]
+async fn call_actor(
+    _lattice_id: String,
```

Comment:
If lattice ID isn't needed here, when would it be useful? I figured that the httpserver couldn't send to anything but the lattice that it belongs to anyways

Reply:
It's needed if the host & capability provider are "multi-tenant"/multi-lattice.